workers/docker-worker/src/lib/util/artifact_download.js

```diff
@@ -38,13 +38,22 @@ module.exports = async function(queue, stream, taskId, artifactPath, destination
     try {
       let expectedSize = 0;
       let receivedSize;
-      let req = request.get(artifactUrl);
+      let req = request.get({
+        method: 'GET',
+        url: artifactUrl,
+        gzip: true,
+      });
       req.on('response', (res) => {
+        // measure the expected HTTP response body size and actual, to compare later
         expectedSize = parseInt(res.headers['content-length']);
         receivedSize = 0;
+        res.on('data', chunk => {
+          // note that this event is emitted with compressed data
+          receivedSize += chunk.length;
+        });
       });
       req.on('data', (chunk) => {
-        receivedSize += chunk.length;
+        // note that this event is emitted with uncompressed data
```

Comment:
Does this portion also run for compressed content, but after uncompression? Otherwise the hash will be empty (maybe `e3b0c44298fc1c1`)

Reply:
Yeah, my understanding is that both `req` and `res` emit `data` events in any case, and that in the content-encoding case one of those is pre- and one is post-decoding.  The tests check an uncompressed artifact as well, verifying this.